### ISSUE TITLE
Update EMA before period end

### ISSUE THREAD

**Issue by claudicular** [DeviaVir/zenbot#1476]
This is a feature request, most of the issue requirements are not relevant it looks like. 

Okay so I understand that for trend_ema the stats used to determine buy/sell only update and trigger an order during the beginning/end of a period. This means that regardless of what is set for poll_trades, no order will be triggered until the end of the period. 

I personally find that for trend_ema, the period needs to be set to be much higher than 2m in order for the algorithm to be profitable, attempting to analyze trends with short periods will end up with a lot of false flags and unprofitable orders. However I think it would be very helpful if when trades are polled it updated the ema value for the latest period so that if a large fall/climb is happening that could trigger an order before the long period is over. 

I can probably write this code and do a pull request but would appreciate if someone can point me to the right direction to make this change, thanks.

**Comment by LoneWolf345** [comment]
Could you try just reducing the period length and increasing the number of periods used for the ema?

**Comment by claudicular** [comment]
Hmmm doesn't seem to work quite as well, maybe giving it time between EMA comparisons is necessary for it to make smarter decisions, if that's the case my feature suggestion is not a good one. If anyone has any other thoughts on this I'd love to hear it. 

What it seems to me is that if it checks for orders too often there will be too many false flags and it'll buy high sell low too often. If it checks for orders too infrequently then it will be too slow on execution. There's some happy medium where you don't do too many orders, do them generally soon enough to capture profit on big swings, and only do a few false flag orders when the price is sideways.

**Comment by Haehnchen** [comment]
i think you mean an indicator thats moving faster on changes because of this dynamic adjustments like WMA or HULL (based on WMA) #1426.

maybe i will provide some strategy about this, iam also interested how this is performs here

**Comment by janus007** [comment]
No, you can't write this code as you think/ suggest. You're asking for something to happen in the future/ now before the interval has ended., This is all fine and dandy but not feasible due to the nature of time and up to the minute in the interval calculations.
I was as the same page as your for decades ago; clear your mind and you will find peace.
